pfcli/configurator/credential.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2021 FriendliAI
+
+"""PeriFlow Credential Configurator."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Tuple
+from uuid import UUID
+
+import typer
+from click import Choice
+from jsonschema import Draft7Validator, ValidationError
+
+from pfcli.configurator.base import InteractiveConfigurator
+from pfcli.service import CredType, ServiceType, cred_type_map_inv
+from pfcli.service.client import (
+    CredentialClientService,
+    CredentialTypeClientService,
+    build_client,
+)
+from pfcli.utils.format import secho_error_and_exit
+
+
+@dataclass
+class CredentialInteractiveConfigurator(InteractiveConfigurator[Tuple[Any, ...]]):
+    """Credential configuration service"""
+
+    ready: bool = False
+    name: Optional[str] = None
+    cred_type: Optional[CredType] = None
+    value: Optional[Dict[str, Any]] = None
+
+    def start_interaction(self) -> None:
+        self.name = typer.prompt(
+            "Enter the name of your new credential.", prompt_suffix="\n>> "
+        )
+        self.cred_type = typer.prompt(
+            "What kind of credential do you want to create?\n",
+            type=Choice([e.value for e in CredType]),
+            prompt_suffix="\n>> ",
+        )
+        cred_type_client: CredentialTypeClientService = build_client(
+            ServiceType.CREDENTIAL_TYPE
+        )
+        schema = cred_type_client.get_schema_by_type(self.cred_type)
+        assert schema is not None
```

Comment:
Actually, this line is not reachable because line 38-41 checks the credential type. I added this line due to the type error. Without this line, line 48 raises a type error (`Object of type "None" is not subscriptable`).